cmd/xl-storage.go

```diff
@@ -1509,6 +1509,16 @@ func (s *xlStorage) ReadVersion(ctx context.Context, volume, path, versionID str
 		}
 	}
 
+	if !skipAccessChecks(volume) && fi.DataDir != "" && !fi.InlineData() {
+		// Verify if the dataDir is present or not when the data
+		// is not inlined to make sure we return correct errors
+		// during HeadObject().
+		_, lerr := Lstat(pathJoin(volumeDir, fi.DataDir))
```

Comment:
you can use 'Access' with F_OK flag; this will not check for permissions and will be faster than Lstat()